src/connection.ts

```diff
@@ -0,0 +1,66 @@
+import { Connection }  from '@libp2p/interface-connection'
+import {ConnectionStat}from '@libp2p/interface-connection'
+import { Stream, Direction } from '@libp2p/interface-connection'
+import { PeerId }  from '@libp2p/interface-peer-id';
+import { AbortOptions }from '@libp2p/interfaces';
+import { logger }      from '@libp2p/logger'
+import { Multiaddr }   from '@multiformats/multiaddr';
+import {v4 as genUuid} from 'uuid';
+
+const log = logger('libp2p:webrtc:connection')
+
+type ConnectionInit = {
+    id: string
+    localPeer: PeerId
+    localAddr?: Multiaddr
+    remotePeer: PeerId
+    remoteAddr: Multiaddr
+    direction: Direction
+    tags?: string[]
+    stat: ConnectionStat
+    pc:  RTCPeerConnection
+    credential_string: string
+}
+
+export class WebRTCConnection implements Connection {
+    id: string;
+    stat: ConnectionStat;
+    remoteAddr: Multiaddr;
+    remotePeer: PeerId;
+    tags: string[] = [];
+    streams: Stream[] = [];
+    direction: Direction;
+
+
+    constructor(init: ConnectionInit) {
+        this.streams = []
+        this.remotePeer = init.remotePeer
+        this.remoteAddr = init.remoteAddr
+        this.stat = init.stat
+        this.id = init.id
+        this.direction = init.direction
+        this.peerConnection = init.pc;
+        this.ufrag = init.credential_string;
+    }
+
+    async newStream(multicodecs: string | string[], options?: AbortOptions): Promise<Stream> {
+        log('TODO',this.ufrag);
+        this.peerConnection.createDataChannel(genUuid());
+        throw new Error("not implemented")
+    }
+
+    addStream(stream: Stream): void {
+        throw new Error("not implemented")
+    }
+    removeStream(id: string): void {
+        throw new Error("not implemented")
+    }
+    async close(): Promise<void> {
+        throw new Error("not implemented")
+    }
+
+    private peerConnection: RTCPeerConnection;
```

Comment:
You may want to move these up to the top of the class with the other class members.

Reply:
Sure thing - whatever the convention is.

I would think I'd want to separate members that exist to satisfy the interface from those that exist for implementation's sake.

Reply:
I don't think I've ever seen that done in TS.